### DISCUSSION TITLE
Fat binary with CUDA backend

### DISCUSSION THREAD

**Discussion by vitduck** [intel/llvm#3903] [Q&A] [answered]
Hi, 

To generate fat binary for multiple sm_*, I have tried two approaches as follow without success: 
```
clang++ 
    -fsycl
    -fsycl-targets=nvptx64-nvidia-cuda-sycldevice
    -Xsycl-target-backend "--cuda-gpu-arch=sm_35" 
    -Xsycl-target-backend "--cuda-gpu-arch=sm_70" 
    vector_add.cpp -o vector_add.x 
```
and 
``` 
clang++ 
    -fsycl
    -fsycl-targets=nvptx64-nvidia-cuda-sycldevice
    -Xsycl-target-backend "--cuda-gpu-arch=sm_35 --cuda-gpu-arch=sm_70" 
    vector_add.cpp -o vector_add.x 
```
I thought the latter were correct. However, the errer was
```
clang-13: error: cannot find libdevice for unknown. 
Provide path to different CUDA installation via --cuda-path, or pass -nocudalib to build without linking with libdevice.
```
Without '-Xsycl-target-backend', it works on V100, but not K40. 
Please kindly advise the correct way to generate fat binary. 
Thanks!

**Comment by alexbatashev** [comment]
[This](https://github.com/intel/llvm/blob/db31088ce3ef5950b98ff16edf8f721724e3b93d/sycl/plugins/cuda/pi_cuda.cpp#L810-L832) piece of code suggests, that at least CUDA plugin is not ready for what you're looking for. I guess, driver support is also limited in this scenario.
Also tagging @steffenlarsen.

**Comment by steffenlarsen** [comment, marked as answer]
There is an open PR for enabling this (https://github.com/intel/llvm/pull/2087).  Sadly I don't think any of the people behind it have the bandwidth to maintain it at this time.

**Reply by vitduck** [reply]
@alexbatashev , @steffenlarsen 
Thanks for the clarification. 
I will adjust the Makefile to accommodate separate SM.
